src/point-cloud-layer.ts

```diff
@@ -0,0 +1,219 @@
+     */
+    pointSize: number;
+
+    /**
+     * Material settings for lighting effect.
+     *
+     * @default true
+     * @see https://deck.gl/docs/developer-guide/using-lighting
+     */
+    material?: Material;
+
+    /** 
+    * Center position accessor.
+    * If not provided, will be inferred by finding a column with extension type
+    * `"geoarrow.point"` or `"geoarrow.multipoint"`.
+    */
+    getPosition?: ga.vector.PointVector | ga.vector.MultiPointVector;
+
+    /**
+     * The normal of each object, in `[nx, ny, nz]`.
+     * @default [0,0,1]
+     */
+    getNormal?: 10 // TODO normalAccessor
+    
+    /**
+     * The rgba color is in the format of `[r, g, b, [a]]`
+     * @default [0,0,0,225] 
+     */
+    getColor?: [0,0,0,225] // TODO colorAccessor
+}
+
+// Remove data nd get Position from the upstream default props
+const {
+    data: _data,
+    getPosition: _getPosition,
+    ..._upstreamDefaultProps
+} = PointCloudLayer.defaultProps;
+
+// Default props added by us
+const ourDefaultProps = {
+    _validate: true,
+};
+
+// @ts-expect-error Type error in merging default props with ours
+const defaultProps: DefaultProps<GeoArrowPointCloudLayerProps> = {
+    ..._upstreamDefaultProps,
+    ...ourDefaultProps,
+}
+
+export class GeoArrowPointCloudLayer<
+    ExtraProps extends {} = {},
+    > extends CompositeLayer<Required<GeoArrowPointCloudLayerProps> & ExtraProps>{
+        static defaultProps = defaultProps;
+        static layerName = "GeoArrowPointCloudLayer";
+
+        getPickingInfo(
+            params: GetPickingInfoParams & {
+                sourceLayer: { props: GeoArrowExtraPickingProps }
+            },
+        ): GeoArrowPickingInfo { 
+            return getPickingInfo(params, this.props.data);
+        }
+
+        renderLayers(): Layer<{}> | LayersList | null {
+            const { data: table } = this.props;
+        
+            const pointVector = getGeometryVector(table, EXTENSION_NAME.POINT);
+            if (pointVector !== null) {
+              return this._renderLayersPoint(pointVector);
+            }
+        
+            const geometryColumn = this.props.getPosition;
+            if (ga.vector.isPointVector(geometryColumn)) {
+              return this._renderLayersPoint(geometryColumn);
+            }
+        
+            throw new Error("geometryColumn not point or multipoint");
+        }
+
+        _renderLayersPoint(
+            geometryColumn: ga.vector.PointVector,
+        ): Layer<{}> | LayersList | null {
+            const { data: table } = this.props;
+            
+            if (this.props._validate) {
+                assert(ga.vector.isPointVector(geometryColumn));
+                validateAccessors(this.props, table);
+            }
+
+            // Exclude manually-set accessors
+            const [accessors, otherProps] = extractAccessorsFromProps(this.props, [
+                "getPosition",
+                "getNormal",
+                "getColors",
```

Comment:
We only need to set `getPosition` manually and can leave `getNormal` and `getColor` (sic) out of this.